mongo/collection.go

```diff
@@ -290,7 +294,13 @@ func (coll *Collection) insert(ctx context.Context, documents []interface{},
 	return result, op.Execute(ctx)
 }
 
-// InsertOne inserts a single document into the collection.
+// InsertOne performs an insert operation (https://docs.mongodb.com/manual/reference/command/insert/) to insert a
+// single document into the collection.
+//
+// The document parameter should be the document to be inserted. It cannot be nil. If the document does not have an
+// _id field when transformed into BSON, one will be added automatically.
```

Comment:
Added a note about the original document not being modified. For omitempty, I purposely left out any behavioral descriptions of BSON marshalling. I don't think it needs to be called out here because `_id` isn't "special" - if the marshalled document doesn't have it, we add an element. The actual marshalling behavior and how omitempty works for various data types should be documented in the bson library.